app/models/service_template/copy.rb

```diff
@@ -6,12 +6,24 @@ def template_copy(new_name = "Copy of " + name + Time.zone.now.to_s)
       ActiveRecord::Base.transaction do
         dup.tap do |template|
           template.update_attributes(:name => new_name, :display => false)
-          service_resources.each do |sr|
-            resource = sr.resource.respond_to?(:service_template_resource_copy) ? sr.resource.service_template_resource_copy : sr.resource
-            template.add_resource(resource, sr)
-          end
+          service_resources.each { |sr| resource_copy(sr, template) }
```

Comment:
Do we need to shorten to `cb`, `cbs` and `sr`? (Especially in the methods below)